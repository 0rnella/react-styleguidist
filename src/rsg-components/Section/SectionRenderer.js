import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Heading from 'rsg-components/Heading';

const styles = ({ space, fontFamily, fontSize }) => ({
	root: {
		marginBottom: space[4],
	},
	heading: {
		margin: [[0, 0, space[2]]],
		fontFamily: fontFamily.base,
		fontSize: fontSize.h1,
	},
});

export function SectionRenderer({ classes, name, slug, content, components, sections }) {
	return (
		<section className={classes.root}>
			{name && (
				<Heading level={1} slug={slug} className={classes.heading}>{name}</Heading>
			)}
			{content}
			{components}
			{sections}
		</section>
	);
}

SectionRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string,
	slug: PropTypes.string,
	content: PropTypes.node,
	components: PropTypes.node,
	sections: PropTypes.node,
};

export default Styled(styles)(SectionRenderer);
