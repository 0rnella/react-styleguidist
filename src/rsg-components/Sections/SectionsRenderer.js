import React, { PropTypes } from 'react';

export default function SectionsRenderer({ sections }) {
	return (
		<div>
			{sections}
		</div>
	);
}

SectionsRenderer.propTypes = {
	sections: PropTypes.array.isRequired,
};
