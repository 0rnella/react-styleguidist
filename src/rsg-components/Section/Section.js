import React, { Component, PropTypes } from 'react';
import Examples from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';

export default class Section extends Component {
	static propTypes = {
		highlightTheme: PropTypes.string.isRequired,
		section: PropTypes.object.isRequired,
		sidebar: PropTypes.bool,
	};

	renderContent(highlightTheme, examples) {
		if (!examples) {
			return null;
		}

		return (
			<Examples highlightTheme={highlightTheme} examples={examples} />
		);
	}

	renderComponents(highlightTheme, components, sections, sidebar) {
		if (!components && !sections) {
			return null;
		}

		return (
			<Components
				highlightTheme={highlightTheme}
				components={components || []}
				sections={sections || []}
				sidebar={sidebar}
			/>
		);
	}

	render() {
		const { highlightTheme, section, sidebar } = this.props;

		return (
			<SectionRenderer
				name={section.name}
				content={this.renderContent(highlightTheme, section.content)}
				components={this.renderComponents(highlightTheme, section.components, section.sections, sidebar)}
			/>
		);
	}
}
