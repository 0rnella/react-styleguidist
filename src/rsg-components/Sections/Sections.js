import React, { Component, PropTypes } from 'react';
import Section from 'rsg-components/Section';
import SectionsRenderer from 'rsg-components/Sections/SectionsRenderer';

export default class Sections extends Component {
	static propTypes = {
		highlightTheme: PropTypes.string.isRequired,
		sections: PropTypes.array.isRequired,
		sidebar: PropTypes.bool,
	};

	renderSections() {
		const { highlightTheme, sections, sidebar } = this.props;

		return sections.map(section => (
			<Section
				key={section.name}
				highlightTheme={highlightTheme}
				section={section}
				sidebar={sidebar}
			/>
		));
	}

	render() {
		return (
			<SectionsRenderer sections={this.renderSections()} />
		);
	}
}
