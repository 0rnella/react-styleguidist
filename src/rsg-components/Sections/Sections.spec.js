import React from 'react';
import noop from 'lodash/noop';
import Section from '../Section';
import Sections from './Sections';
import SectionsRenderer from './SectionsRenderer';

const sections = [
	{
		name: 'Foo',
		content: [
			{
				type: 'code',
				content: '<button>OK</button>',
				evalInContext: noop,
			},
		],
		components: [],
	},
	{
		name: 'Bar',
		content: [
			{
				type: 'markdown',
				content: 'Hello *world*!',
			},
		],
		components: [],
	},
];

it('should render component renderer', () => {
	const actual = shallow(
		<Sections sections={sections} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();
});

it('render should render component', () => {
	const actual = shallow(
		<SectionsRenderer
			sections={[
				<Section key={1} section={sections[0]} />,
				<Section key={2} section={sections[1]} />,
			]}
		/>
	);

	expect(shallowToJson(actual)).toMatchSnapshot();
});
