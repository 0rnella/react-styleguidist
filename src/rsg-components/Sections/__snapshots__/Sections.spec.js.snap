exports[`test render should render component 1`] = `
<div>
  <Section
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
        ],
        "name": "Foo",
      }
    } />
  <Section
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ],
        "name": "Bar",
      }
    } />
</div>
`;

exports[`test should render component renderer 1`] = `
<SectionsRenderer
  sections={
    Array [
      <Section
        section={
            Object {
                "components": Array [],
                "content": Array [
                  Object {
                    "content": "<button>OK</button>",
                    "evalInContext": [Function],
                    "type": "code",
                  },
                ],
                "name": "Foo",
              }
        } />,
      <Section
        section={
            Object {
                "components": Array [],
                "content": Array [
                  Object {
                    "content": "Hello *world*!",
                    "type": "markdown",
                  },
                ],
                "name": "Bar",
              }
        } />,
    ]
  } />
`;
